src/main/java/cn/nukkit/item/Item.java

```diff
@@ -395,10 +391,22 @@ public static Item get(int id, Integer meta, int count, byte[] tags) {
             if (c == null) {
                 item = new Item(id, meta, count);
             } else if (id < 256) {
-                if (meta >= 0) {
-                    item = new ItemBlock(Block.get(id, meta), meta, count);
+                if (customblocklist[id] == null) {
+                    if (meta >= 0) {
+                        item = new ItemBlock(Block.get(id, meta), meta, count);
+                    } else {
+                        item = new ItemBlock(Block.get(id), meta, count);
+                    }
                 } else {
-                    item = new ItemBlock(Block.get(id), meta, count);
+                   try{
+                    item = ((ItemBlock) customblocklist[id].getConstructor(Block.class, Integer.class, int.class).newInstance(Block.get(id, meta), meta, count));
+                }catch (Exception e){
```

Comment:
no.

you should check whether or not the class has a valid constructor when it's being registered (in `registerCustomItemBlock`) and then store the constructor in the registry, not have to look up the constructor every time. implementations should be expected to provide a valid constructor, and if an exception occurs in the constructor (`InstantiationException`) then we should fail, not silently ignore it and work around.

Reply:
I think that keeping the Vanilla Classes and Custom Classes should be seperate. So i dont think that Putting the Custom Item Block Classes in  List would be the best... I guess it would a tad faster.